close #16 회원가입 + 로그인

Jennorresothie opened the pull request:
close #16 회원가입 + 로그인 구현

Jennorresothie committed:
feat(BE) : UserEntity 생성

Jennorresothie committed:
feat(BE) : 회원가입 service 구현, 아이디 중복 예외

회원가입 Service단 메소드 구현
ID 중복 체크 예외 생성
UpsertUser dto 생성

Jennorresothie committed:
feat(BE) : 회원가입 controller 구현, userrepository 메소드 수정,  비밀번호 암호화, dto 추가

회원가입 controller 구현
userrepository 메소드 수정
비밀번호 암호화를 위한 passwordEncoder
UpsertUserRequest dto 추가

Jennorresothie committed:
feat(BE) : 로그인 Service 구현, 회원 예외 처리

로그인 Service 구현
회원 예외 처리를 위한 UserNotFoundException 추가

Jennorresothie committed:
feat(BE) : 로그인 Controller 구현

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/service/UserService.java:
```diff
@@ -1,4 +1,43 @@
+    public boolean signup(UpsertUser userData) {
+        // email 중복 체크
+        UserEntity user = userRepository.findByEmailAndIsDeletedIsFalse(userData.getEmail())
+                .orElseThrow( ()->  new AlreadyExistEmailException(userData.getEmail()));
```
orElseThrow는 null일때 작동하지 않나요?

jhchoi57 replied:
isPresent로 수정

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/repository/exception/AlreadyExistEmailException.java:
```diff
@@ -0,0 +1,7 @@
+package com.jootcamp.superboard.user.repository.exception;
+
+public class AlreadyExistEmailException extends RuntimeException{
```
BadRequestException 상속 + GlobalExceptionHandler에서 처리 필요

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/service/UserService.java:
```diff
@@ -1,4 +1,43 @@
+        UserEntity userEntity = userRepository.findByEmailAndIsDeletedIsFalse(user.getEmail())
+                .orElseThrow(()->new UserNotFoundException(user.getEmail()));
+
+        if(user.getPassword()==userEntity.getPassword())
```
== 말고 equals로 수정

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/controller/UserApiController.java:
```diff
@@ -0,0 +1,53 @@
+            session.setMaxInactiveInterval(1800); // Session이 30분동안 유지
+
+            return ResponseEntity.status(HttpStatus.OK).build();
+        }
```
service layer로 로직 옮겨주세요

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/controller/UserApiController.java:
```diff
@@ -0,0 +1,53 @@
+    //회원가입
+    @PostMapping("/signup")
+    public ResponseEntity<Boolean> signup(@RequestBody UpsertUserRequest userRequest) throws Exception {
+        String encodingPass = passwordEncoder.encrypt(userRequest.getPassword());
```
encrypt 과정도 service에서 해주세요

Jennorresothie committed:
fix(BE) : 요청 사항 수정

- isPresent로 수정
- BadRequestException 생성 및 상속
- == 말고 equals로 수정
- encrypt 과정 및 세션 생성과정 service layer 이전

Jennorresothie committed:
feat(BE) : 로그인 확인 필터

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/service/UserService.java:
```diff
@@ -1,4 +1,63 @@
+        // email 중복 체크
+        Optional<UserEntity> user = userRepository.findByEmailAndIsDeletedIsFalse(userData.getEmail());
+        String encodingPass = passwordEncoder.encrypt(userData.getPassword());
+        if(user.isPresent()) throw new AlreadyExistEmailException(userData.getEmail());
```
ifPresent 해보고 안되면 장 규민씨에게 질문

jhchoi57 commented on src/main/java/com/jootcamp/superboard/common/filter/LoginCheckFilter.java:
```diff
@@ -0,0 +1,49 @@
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+        try {
+            if (isLoginCheckPath(requestURI)) {
```
OncePerRequestFilter 상속받고 shouldNotFilter 적용

jhchoi57 commented on src/main/java/com/jootcamp/superboard/common/filter/LoginCheckFilter.java:
```diff
@@ -0,0 +1,49 @@
+                if (session == null || session.getAttribute("userId") == null) {
+                    System.out.println("미인증 사용자 요청 {}" + requestURI);
+                    //로그인으로 redirect
+                    //httpResponse.sendRedirect("/login?redirectURL=" + requestURI);
```
로그인 상태가 아니니까 401 내려주고
프론트에서 처리
